src/eos-updater-fetch.c

```diff
@@ -72,6 +74,31 @@ update_progress (OstreeAsyncProgress *progress,
     eos_updater_set_downloaded_bytes (updater, bytes);
 }
 
+static gboolean
+repo_pull (OstreeRepo *self,
+           const gchar *remote_name,
+           const gchar *ref,
+           const gchar *url_override,
+           OstreeAsyncProgress *progress,
+           GCancellable *cancellable,
+           GError **error)
+{
+  g_auto(GVariantBuilder) builder = { { { 0, } } };
+  g_autoptr(GVariant) options = NULL;
+
+  g_variant_builder_init (&builder, G_VARIANT_TYPE ("a{sv}"));
+  g_variant_builder_add (&builder, "{s@v}", "refs",
+                         g_variant_new_variant (g_variant_new_strv (&ref, 1)));
+  if (url_override != NULL)
+    g_variant_builder_add (&builder, "{s@v}", "override-url",
+                           g_variant_new_variant (g_variant_new_string (url_override)));
+
+  options = g_variant_ref_sink (g_variant_builder_end (&builder));
+  return ostree_repo_pull_with_options (self, remote_name, options,
+                                        progress, cancellable, error);
```

Comment:
Actually, I think the Kinvolk branch does this already, and we *don’t* want to enable `gpg-verify-summary`, because that means that pulling from a LAN peer requires the peer to have a valid `/ostree/repo/summary` file — which it might not have. The `summary` file is not pulled from the server by default, and even if it is, the user could locally nuke it using `ostree summary` (though that’s unlikely).

The code they have instead is in `eos-updater-poll-common.c`, in `commit_checksum_from_*()`, called by `fetch_commit_checksum()`. It works its way through the priority list of `refs.d`, `eos-summary`, original `summary`, and verifies them and tries to look up the given ref in them. The verification code looks the same as what `ostree.git` uses when you pass `gpg-verify-summary`.

The repository mirroring code currently copies the `refs.d` and `eos-summary` files (and their signatures). If no `eos-summary` file is available on the remote, but a normal `summary` file is, the `extensions->summary = …` assignment at the bottom of `commit_checksum_from_any_summary()` should ensure it gets copied to the mirror as `eos-summary`. Same for all the signatures.